src/qemu.rs

```diff
@@ -935,12 +938,44 @@ impl Qemu {
         Self::stream_child_output(self.updates.clone(), &mut child);
 
         // Ensure child is cleaned up even if we bail early
-        let child = scopeguard::guard(child, Self::child_cleanup);
+        let mut child = scopeguard::guard(child, Self::child_cleanup);
 
         if let Err(e) = self.wait_for_qemu() {
             return Err(e).context("Failed waiting for QEMU to be ready");
         }
 
+        // Connect to QMP socket
+        let qmp_unix_stream = match self.connect_to_uds(&self.qmp_sock) {
+            Ok(s) => s,
+            Err(e) => {
+                let err = Self::extract_child_stderr(&mut child);
+                return Err(e).context("Failed to connect QMP").context(err);
+            }
+        };
+
+        // UnixStream is not directly cloneable, so we cannot use `Qmp::from_stream`
+        // without borrowoing the stream. Instead we create a new stream from the
+        // qmp_unix_stream and pass it into Qmp::new, which takes ownership of the
+        // stream.
+        let qmp_stream = qapi::Stream::new(
+            BufReader::new(
+                qmp_unix_stream
+                    .try_clone()
+                    .context("Failed to clone QMP Unix Stream")?,
+            ),
+            qmp_unix_stream,
+        );
+        let mut qmp = Qmp::new(qmp_stream);
```

Comment:
This is b/c we can't pass a reference to the `UnixStream` like we did before cuz now `qmp` outlives the `UnixStream`, right?

Reply:
correct.